react-app/src/app/screens/User/components/History/index.js

```diff
@@ -1,25 +1,63 @@
 import React, { Component } from 'react';
-// import PropTypes from 'prop-types';
-// import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+import { getUserPoints } from '../../../../../redux/Game/actions';
+
+import style from './styles.scss';
+import Snapshot from './Snapshot';
 
-// TODO: implementate history user
 class History extends Component {
+  componentDidMount() {
+    this.props.getUserPoints(this.props.token);
+  }
+  renderHistoryPoints = () => {
+    if (this.props.historyPoints) {
+      const keysSnapshots = Object.keys(this.props.historyPoints);
+      const snapshots = keysSnapshots.map((key, i) => {
+        const date = new Date(Number(key));
+        return (
+          <Snapshot
+            key={i.toString()}
+            date={date.toLocaleDateString()}
+            points={this.props.historyPoints[key]}
+          />
+        );
+      });
+      return snapshots;
+    }
+    return null;
+  };
   render() {
-    return <h3>History</h3>;
+    return (
+      <div className={style.history}>
+        <h3>History</h3>
+        <div className={style.historySnapshots}>{this.renderHistoryPoints()}</div>
```

Comment:
```suggestion
        {this.props.historyPoints && <div className={style.historySnapshots}>{this.renderHistoryPoints()}</div>}
```
This way you would simplify the points renderer method and only call it when there are points to render